### PR TITLE
Add some tests for invalid subtyping due to non-matching field types

### PR DESCRIPTION
**Pull request by fitzgen**
Specifically around the requirements that both field types have the same mutability and that the storage type must be exactly the same when the field type is mutable.

@tlively PTAL, thanks!

### CODE REVIEW

File: test/core/gc/type-subtyping-invalid.wast
Comment (rossberg) [inline test/core/gc/type-subtyping-invalid.wast:25]: Perhaps keep the actual storage types the same in this test (and the next one)?

**Review by rossberg — COMMENTED** [review]
Thanks! Why not include these in the existing type-subtyping?

File: test/core/gc/type-subtyping-invalid.wast
Comment (fitzgen) [reply test/core/gc/type-subtyping-invalid.wast:25]: > Perhaps keep the actual storage types the same in this test (and the next one)?

Whoops, yeah that was the intention, just a copy-paste issue. Will push a fix in a moment.

File: test/core/gc/type-subtyping-invalid.wast
Comment (tlively) [inline test/core/gc/type-subtyping-invalid.wast:16]: It looks like this test intends to have constant fields, but in fact it has mutable fields. Do we already test the case where the subtyping is correctly covariant, but the fields are mutable so it is still invalid?

**Review by tlively — APPROVED** [review]

**Comment by fitzgen** [comment]
Thanks for the approve @tlively! FWIW, I do not have merge permissions on this repo, so I can't merge this PR myself.

**Comment by rossberg** [comment]
I regularly go through the proposal repos and merge them into wasm-3.0, so new commits should eventually make it there. I haven't done any back-merging, because that is way to tedious, and at this point I'm not sure how much it still matters.

**Comment by tlively** [comment]
Sounds good, I'll go ahead and merge this, then.

**Pull request merged**
